src/Shared/FileMatcher.cs

```diff
@@ -2277,6 +2300,53 @@ private static int SkipSlashes(string aString, int startingIndex)
             return index;
         }
 
+        /// <summary>
+        /// Returns true if drive enumerating wildcard patterns are detected using the directory and wildcard parts.
+        /// </summary>
+        private static bool IsDriveEnumeratingWildcardPattern(string directoryPart, string wildcardPart)
```

Comment:
Could you solve some of the test problems by making this `internal` and testing it directly instead of having to plumb real-looking paths all the way through the real system?